pom.client.xml

```diff
@@ -1073,6 +1073,50 @@
       </build>
     </profile>
 
+    <!-- Skip codesnippet injection unless -Dinject-codesnippets is passed as an argument.
+      This is specifically done to prevent requiring python as a pre-requisite since
+      this is only needed by the build system. This is also necessary due to limitations in consumption
+      from docs side. 
+       -->
+    <profile>
+      <id>replace-codesnippet-references</id>
+      <activation>
+        <property>
+          <name>inject-codesnippets</name>
+        </property>
+        <file>
+          <exists>../../../eng/pipelines/scripts/inject_codesnippets.py</exists>
+        </file>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>1.2.1</version> <!-- {x-version-update;org.codehaus.mojo:exec-maven-plugin;external_dependency} -->
+            <executions>
+              <execution>
+                <id>replace-codesnippet-references</id>
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <executable>python</executable>
+              <workingDirectory>${project.basedir}</workingDirectory>
+              <arguments>
+                <argument>../../../eng/pipelines/scripts/inject_codesnippets.py</argument>
```

Comment:
I'm not a big fan of calling python from maven but we already do it and you made it opt-in so I'll live with it.